### PR TITLE
Create hpc-stack environment to match UFS development versions

### PR DESCRIPTION
**Pull request by kgerheiser**
@jkbk2004 @ulmononian @BrianCurtis-NOAA I have a hpc-stack equivalent of Spack here if you want to take a look. Then, we can extend it to other machines.

The versions are mostly the same except for 

* Zlib 1.2.12 is used because earlier versions have been deprecated
* PIO 2.5.7 is used to support external PIO option in ESMF added in 8.3.0

Some notable additions are MET/METplus, METIS/ParMETIS, and ufspyenv bundle for some commonly used Python packages.

See this for some differences between Spack and hpc-stack modules:

https://github.com/NOAA-EMC/spack-stack/wiki/Transitioning-to-Spack-from-hpc-stack

There is a test installation I added on Orion at:

```
module use module use /work/noaa/da/jedipara/spack-stack/modulefiles
module load miniconda/3.9.7
module use /work/noaa/nems/gkyle/spack-stack/envs/hpc-stack-dev-test/install/modulefiles/Core
module load stack-intel/2022.0.2
module load stack-intel-oneapi-mpi/2021.5.1
module load stack-python/3.9.7
```

I tested this using UFS_UTILS chgres_cube tests and the diffs were very small:

```
Variable Group Count          Sum      AbsSum          Min         Max       Range         Mean      StdDev
w        /       955  1.40238e-09 4.53011e-08 -7.45058e-09 3.72529e-09 1.11759e-08  1.46846e-12 3.31917e-10
zh       /         1  3.05176e-05 3.05176e-05  3.05176e-05 3.05176e-05           0  3.05176e-05           0
t        /         6   0.00012207 0.000183105 -3.05176e-05 3.05176e-05 6.10352e-05  2.03451e-05 2.49175e-05
delp     /         2 -0.000366211 0.000366211 -0.000244141 -0.00012207  0.00012207 -0.000183105 8.63167e-05
sphum    /         2 -3.18323e-12 4.09273e-12 -3.63798e-12 4.54747e-13 4.09273e-12 -1.59162e-12 2.89399e-12
liq_wat  /         4 -2.84044e-14 2.84391e-14 -2.84217e-14 1.04083e-17 2.84321e-14 -7.10109e-15 1.42137e-14
ice_wat  /         5  2.22109e-16 2.22109e-16  2.11758e-22 2.22045e-16 2.22044e-16  4.44219e-17 9.92941e-17
snowwat  /        11 -1.48904e-11 4.40565e-11 -2.91038e-11 1.45519e-11 4.36557e-11 -1.35367e-12 1.01916e-11
graupel  /         1  5.55112e-17 5.55112e-17  5.55112e-17 5.55112e-17           0  5.55112e-17           0
u_w      /        63 -1.05911e-06 1.71646e-06 -4.76837e-07 1.19209e-07 5.96046e-07 -1.68113e-08  9.2701e-08
v_w      /        67  5.71255e-07 2.93117e-06 -4.76837e-07 9.53674e-07 1.43051e-06  8.52619e-09 1.45774e-07
u_s      /        53  9.79395e-07 1.29593e-06 -5.96046e-08 4.76837e-07 5.36442e-07  1.84792e-08 8.05106e-08
v_s      /        86  -5.4026e-07 4.33761e-06 -9.53674e-07 4.76837e-07 1.43051e-06  -6.2821e-09 1.55428e-07
```

Then, I ran a ufs-weather-model control_c48 test which ran to completion (output here `/work/noaa/stmp/gkyle/stmp/gkyle/FV3_RT/rt_39745`)

The modules I used are here:

https://github.com/kgerheiser/ufs-weather-model/blob/spack-stack/modulefiles/ufs_orion.intel
https://github.com/kgerheiser/ufs-weather-model/blob/spack-stack/modulefiles/ufs_common

https://github.com/kgerheiser/UFS_UTILS/blob/spack-stack/modulefiles/build.orion.intel.lua

### CODE REVIEW

File: configs/templates/hpc-stack-dev/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+      version: [1.5.0]
+    grib-util:
+      version: [1.2.3]
+    gsl-lite:
+      version: [0.37.0]
+    hdf:
+      version: [4.2.15]
+      variants: ~fortran ~netcdf
+    hdf5:
+      version: [1.10.6]
```
Comment (climbfuji) [inline configs/templates/hpc-stack-dev/spack.yaml:78]: JEDI is definitely beyond that version, don't think we'll want to use that.

File: configs/templates/hpc-stack-dev/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    yafyaml:
+      version: [0.5.1]
+    zlib:
+      version: [1.2.12]
+
+  definitions:
+  # Set compilers here
+  - compilers: ['%intel']
+  - packages:
+      - global-workflow-env@hpc-stack-dev
```
Comment (climbfuji) [inline configs/templates/hpc-stack-dev/spack.yaml:215]: I didn't know that this syntax works, interesting.

**Review by climbfuji — COMMENTED** [review]
Is the purpose of this hpc-stack-dev to try to replicate the current environment/modules, verify that it's all good and then move to the newer versions (e.g. `hdf5@1.12.x`). JEDI won't be able to use this template, because we require newer versions of hdf5 than used here. But as long as this gets installed as a separate environment and we use something else going forward (not skylab either), then that's totally fine with me.

**Comment by kgerheiser** [comment]
> Is the purpose of this hpc-stack-dev to try to replicate the current environment/modules, verify that it's all good and then move to the newer versions (e.g. hdf5@1.12.x).

Yes. I'm all for doing it at once, but it was suggested to create an environment that matches ufs-weather-model develop first (@jkbk2004).

File: configs/templates/hpc-stack-dev/spack.yaml (hunk 2 above)
Comment (kgerheiser) [reply configs/templates/hpc-stack-dev/spack.yaml:215]: The matrix or the version? It's useful to declare compilers as a list. If you want to build with multiple compilers you can just add to the list `spack add -l compilers "%gcc"`.

Comment (climbfuji) [reply configs/templates/hpc-stack-dev/spack.yaml:215]: I was referring to the `@hpc-stack-dev` syntax, haven't seen that before.

Comment (kgerheiser) [reply configs/templates/hpc-stack-dev/spack.yaml:215]: The bundles have no source associated with their version so you can give whatever version you want. I think it's good to differentiate bundle versions per environment.

Comment (climbfuji) [reply configs/templates/hpc-stack-dev/spack.yaml:215]: Yes, makes total sense.

**Review by climbfuji — APPROVED** [review]

**Pull request merged**